lib/spack/spack/spec.py

```diff
@@ -188,8 +188,10 @@
                  '^': dependency_color,
                  '#': hash_color}
 
-"""Regex used for splitting by spec field separators."""
-_separators = '[%s]' % ''.join(color_formats.keys())
+"""Regex used for splitting by spec field separators.
+These need to be escaped to avoid metacharacters in color_formats.keys().
+"""
```

Comment:
So, I kind of like `#:`, but it's a `numpy` style docstring, not a google style one.  Should we use `#:`, or be consistent with the after-variable docstrings?  FWIW, neither of these actually add the `__doc__` attribute like for functions -- they're Sphinx/napoleon things, not Python things.